### PR TITLE
Closes #1205 Enable listing and using templates from packages other than {admiral}

### LINKED ISSUE #1205: Feature Request: Allow using and listing template from {admiral} extension packages
### Feature Idea

We need a way for `list_all_templates()` and `use_ad_template()` to specify in which package to look for the templates. Right now it's restricted to templates that ship with {admiral} itself which is not sufficient going forward as extension packages will contain their own templates.

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

I could think of the following:

```
list_all_templates(package = "admiralxxx")
use_ad_template("adxx", package = "admiralxxx")
```

where `package` is `"admiral"` by default but can be changed to other packages.

### PR DESCRIPTION
**Pull request by thomas-neitmann**
Close #1205 

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by thomas-neitmann** [comment]
@rossfarrugia Would be great if you could check whether this is actually working with the {admiralonco} templates.

**Comment by rossfarrugia** [comment]
@thomas-neitmann absolutely, will do... once we actually have one to test with that is! :) should be in the next week or 2 as just waiting on the required new functions to merge to devel.

**Comment by bms63** [comment]
Make sure to put Closes #1205 so it auto closes when we merge into main

File: R/user_helpers.R [hunk 1]
```diff
@@ -73,8 +81,15 @@ use_ad_template <- function(adam_name = "adsl",
 #'
 #' @examples
 #' list_all_templates()
-list_all_templates <- function() {
-  list.files(system.file("templates", package = "admiral")) %>%
+list_all_templates <- function(package = "admiral") {
```
Comment (bundfussr) [inline R/user_helpers.R:84]: If no templates are found, the output is
```
Existing templates:
• 
```
I would prefer something like "No templates available in package xxx".

Comment (thomas-neitmann) [reply R/user_helpers.R:84]: Agree.

Comment (thomas-neitmann) [reply R/user_helpers.R:84]: Now looking like this:

```r
list_all_templates("admiral")
```
```
Existing ADaM templates in package 'admiral':
• ADAE
• ADCM
• ADEG
• ADEX
• ADLB
• ADPP
• ADSL
• ADVS
```
and
```r
list_all_templates("dplyr")
```
```
No ADaM templates available in package 'dplyr'
```

**Review by bundfussr — APPROVED** [review]

**Pull request merged**